compiler/rustc_mir_transform/src/check_unsafety.rs

```diff
@@ -600,13 +615,41 @@ pub fn check_unsafety(tcx: TyCtxt<'_>, def_id: LocalDefId) {
 
                 err.emit();
             }
-            UnsafetyViolationKind::UnsafeFn => tcx.struct_span_lint_hir(
-                UNSAFE_OP_IN_UNSAFE_FN,
-                lint_root,
-                source_info.span,
-                format!("{} is unsafe and requires unsafe block (error E0133)", description,),
-                |lint| lint.span_label(source_info.span, description).note(note),
-            ),
+            UnsafetyViolationKind::UnsafeFn => match details.help_missing_feature(tcx) {
+                Some(help) => match details.note_missing_feature(tcx) {
+                    Some(note_missing_feature) => tcx.struct_span_lint_hir(
+                        UNSAFE_OP_IN_UNSAFE_FN,
+                        lint_root,
+                        source_info.span,
+                        format!(
+                            "{} is unsafe and requires unsafe block (error E0133)",
+                            description,
+                        ),
+                        |lint| {
+                            lint.span_label(source_info.span, description)
+                                .help(help)
+                                .note(note_missing_feature)
+                        },
+                    ),
+                    None => tcx.struct_span_lint_hir(
```

Comment:
This too, is still way too verbose. It repeats the same thing three times. Ideally you'd only have one `struct_span_lint_hir` call and move the match on details into the closure.

Reply:
Done.